File: LeetCode/1482-minimum-number-of-days-to-make-m-bouquets/1482-minimum-number-of-days-to-make-m-bouquets.cpp
class Solution {
public:
    int getNumOfBouquets(vector<int> &bloomDay, int mid, int k) {
        int numOfBouquets = 0;
        int consecutiveCount = 0;
        for (int i = 0; i < bloomDay.size(); i++) {
            if (bloomDay[i] <= mid) {
                consecutiveCount++;
            } else {
                consecutiveCount = 0;
            }
            if (consecutiveCount == k) {
                numOfBouquets++;
                consecutiveCount = 0;
            }
        }
        return numOfBouquets;
    }

    int minDays(vector<int>& bloomDay, int m, int k) {
        int startDay = 0;
        int endDay = *max_element(bloomDay.begin(), bloomDay.end());
        int minDays = -1;
        while (startDay <= endDay) {
            int mid = startDay + (endDay - startDay) / 2;
            if (getNumOfBouquets(bloomDay, mid, k) >= m) {
                minDays = mid;
                endDay = mid - 1;
            } else {
                startDay = mid + 1;
            }
        }
        return minDays;
    }
};

File: LeetCode/1482-minimum-number-of-days-to-make-m-bouquets/1482-minimum-number-of-days-to-make-m-bouquets.py
class Solution:
    def getNumOfBouquets(self, bloomDay: List[int], mid: int, k: int) -> int:
        numOfBouquets = 0
        consecutiveCount = 0
        for day in bloomDay:
            if day <= mid:
                consecutiveCount += 1
            else:
                consecutiveCount = 0
            if consecutiveCount == k:
                numOfBouquets += 1
                consecutiveCount = 0
        return numOfBouquets

    def minDays(self, bloomDay: List[int], m: int, k: int) -> int:
        startDay = 0
        endDay = max(bloomDay)
        minDays = -1
        while startDay <= endDay:
            mid = startDay + (endDay - startDay) // 2
            if self.getNumOfBouquets(bloomDay, mid, k) >= m:
                minDays = mid
                endDay = mid - 1
            else:
                startDay = mid + 1
        return minDays